### PR TITLE
NoteListViewController: Implementing UIContextualActions

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details
- Updates **UITableViewRowAction > UIContextualAction**
- Updates several assets!
- New **Copy Internal Link** Swipeable Action
- Added missing Tracker invocations

@aerych sir!! may I bug you yet again?
Thaaaank youuu!

Ref. #863

### Test: Interlink
1. Open the Notes List
2. Swipe over any row
3. Press the Link button (purple!)
4. Press over any other note

- [x] Verify that you can **paste** a Markdown Link to the selected note
- [x] Verify that the Markdown Link looks as follows: `[Title](simplenote://note/HASH)`

### Test: Regressions
- [x] Verify you can Pin / Unpin a Note by swiping over its row (Notes List!)
- [x] Verify you can Trash a Note by swiping over its row (Notes List!)
- [x] Verify you can Share a Note by swiping over its row (Notes List!)

### Release
`RELEASE-NOTES.txt` was updated in d2c3ae3 with:
 
> Swiping a Note in the List now reveals the Copy Internal Link Action

### CODE REVIEW

File: Simplenote/Classes/SPNoteListViewController+Extensions.swift
Comment (aerych) [inline Simplenote/Classes/SPNoteListViewController+Extensions.swift:585]: Maybe a little nitpicky of me, but I'm wondering... is it easy to miss what this is doing given the statement is spread out over multiple lines?  Would adding a `return` keep things clear?

File: Simplenote/Note+Interlink.swift
Comment (aerych) [inline Simplenote/Note+Interlink.swift:15]: It looks like this approach can truncate a title in the middle of a word. Maybe not a big deal unless you have a very long title, but would it make sense to check for the nearest word break to the target length and truncate from there?  It would avoid chopping a word in the middle. Maybe the fixed length as a fallback when there are no spaces within a reasonable distance? 
Just a thought.

**Review by aerych — APPROVED** [review]
Hola @jleandroperez 👋 
Looks good! Tested as described and everything worked as expected.  I had two nitpicky suggestions code wise. Please take or leave them as you like.  Otherwise :shipit: !

File: Simplenote/Classes/SPNoteListViewController+Extensions.swift
Comment (jleandroperez) [reply Simplenote/Classes/SPNoteListViewController+Extensions.swift:585]: Good point, done, thank you!

**Comment by jleandroperez** [comment]
@aerych You're a genius, sir!! Title truncation mechanism implemented. Mind taking another peek?

**Comment by aerych** [comment]
Love it! :shipit:  sir!

**Comment by mokagio** [comment]
I'm gonna go ahead and merge this one. It's code freeze time for 4.24 and I don't see anything in the conversation and commits after the PR was approved pointing to wait to merge this.

My best guess is that it wasn't merged because we were waiting on CI to finish after the merge of `develop` into this. CI's green now.

**Comment by jleandroperez** [comment]
**Thank you Gio!!!** I was holding a bit to sync the launch with other platforms, BUT (if needed!) we'll just disable the feature in the release branch.

**Comment by mokagio** [comment]
Oh, no worries... I should have waited 😳 

Well, if we need to disable the feature flag, just let me know and I'll do it 👍

**Comment by jleandroperez** [comment]
Thank you @mokagio !!!

No worries, we're pushing this back to 4.25 (PR #873). I'll ping you as soon as it's merged! 😄

**Pull request merged**
